src/components/pages/ManagementRegime/ManagementRegime.js

```diff
@@ -209,6 +209,7 @@ const ManagementRegime = ({ isNewManagementRegime }) => {
   const [isDeletingRecord, setIsDeletingRecord] = useState(false)
   const [isDeleteRecordModalOpen, setIsDeleteRecordModalOpen] = useState(false)
   const [currentDeleteRecordModalPage, setCurrentDeleteRecordModalPage] = useState(1)
+  const shouldPromptTrigger = isFormDirty && saveButtonState !== buttonGroupStates.saving // we need to prevent the user from seeing the dirty form prompt when a new MR is saved (and that triggers a navigation to its new page)
 
```

Comment:
The order of state changes seems different. More race-condition-y and finicky, or just differently race-condition-y